docs/CODE_GUIDELINES.md

```diff
@@ -198,80 +217,196 @@ exampleClass      ->InitExample();
-biggerExampleClass = new CBiggerExampleClass(value1, value2);
-exampleClass->InitExample();
-biggerExampleClass->InitExample();
+std::vector<std::string> test; test.push_back("foobar");
 ```
 
 **[back to top](#table-of-contents)**
 
-## 10. Controls Statements
-Insert a new line before every:
-* else in an if statement
-* catch in a try statement
-* while in a do statement
+## 5. Declarations
 
-### 10.1. if else
-Put `then`, `return` or `throw` statements on a new line. Keep `else if` statements on one line.
+### 5.1. Multiple declarations
+Do not put multiple declarations on a single line. This avoids confusion with differing pointers, references, and initialization values on the same line (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es10-declare-one-name-only-per-declaration)).
+
+✅ Good:
 ```cpp
-if (true)
-  return;
-if (true)
-{
-  [...]
-} 
-else if (false)
+char* a;
+char b;
+```
+
+❌ Bad:
+```cpp
+char* a, b;
+```
+
+### 5.2. Pointer and reference types
+Left-align `*` and `&` to the base type they modify.
+
+✅ Good:
+```cpp
+char* a;
+void test(const std::string& b);
+```
+
+❌ Bad:
+```cpp
+char *a;
+char * b;
+void test(const std::string &b);
+```
+
+### 5.3. `const` and other modifiers
+Place `const` and similar modifiers in front of the base type.
+
+✅ Good:
+```cpp
+void test(const std::string& a);
+```
+
+❌ Bad:
+```cpp
+void test(std::string const& a);
+```
+
+### 5.4. Initialization
+
+Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor. Zero-initialize with `{}`.
+
+✅ Good:
+```cpp
+int x{};
+int* y{};
+CLog::Log("test: {} {}", x, y);
+```
+
+❌ Bad:
+```cpp
+int x; // used uninitialized
+int* y = nullptr; // default-initialization not using {}
+CLog::Log("test: {} {}", x, y);
+```
+
+In general, prefer the `{}` initializer syntax over alternatives. This syntax is less ambiguous and disallows narrowing (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list)).
+
+✅ Good:
+```cpp
+int x{5};
+int y{x};
+```
+
+**[back to top](#table-of-contents)**
+
+## 6. Scoping
+
+### 6.1. Namespaces
+
+Try to put all code into appropriate namespaces (e.g. following directory structure) and avoid polluting the global namespace.
+
+### 6.2. Local functions
+
+Put functions local to a compilation unit into an anonymous namespace.
+
+✅ Good:
+```cpp
+namespace
 {
-  return;
-} 
-else
-  return;
+
+void test();
+
+}
 ```
 
-### 10.2. switch case
+❌ Bad:
 ```cpp
-switch (cmd)
+static void test();
+```
+
+**[back to top](#table-of-contents)**
+
+## 7. Headers
+Included header files have to be sorted alphabetically to prevent duplicates and allow better overview, with an empty line clearly separating sections.
+
+Header order has to be:
+* Own header file
+* Other Kodi includes
+* C and C++ system files
+* Other libraries' header files
```

Comment:
We generally put other libraries' headers before system files

Reply:
Can we clarify if sort order is case sensitive or not. The example implies a...z,A...Z, but good to state it explicitly if that is what we want. I don't mind which and I have seen both in existing code (or maybe it was just a jumble).

Reply:
> Can we clarify if sort order is case sensitive or not. The example implies a...z,A...Z,

I don't think this is implied by the example. It explicitly says: "Place directories before files", which is I think the rule that is followed. Do we actually have cases where case matters (pun not intended) after putting dirs first?

Reply:
Doh, I have clearly never understood what "Place directories before files" meant! But that does make sense of what I have seen, sorry for noise